### PR TITLE
Add merge indicator to fork choice struct

### PR DESCRIPTION
**Pull request by realbigsean**
## Issue Addressed

- The merge P2P spec requires checking whether a block's parent state has a non-empty `ExecutionPayload`.  In order to avoid loading a parent state for every block we see in gossip, this PR adds an `is_merge_complete` `bool` to the `ProtoNode` struct
 so we can call `fork_choice.get_block(block.parent_root).is_merge_complete`

## Proposed Changes

- Adds the field described above
- Adds a database migration from schema 4 to schema 5, so we can add the new field to the `PersistedForkChoice`

## Additional Info

To facilitate the database migration I added the new field to `ProtoNode` and created a `LegacyProtoNode` struct (which doesn't include the new field) and `LegacySszContainser` struct. I initially tried to use `superstruct` to create variants with and without the field but it didn't seem worth the complexity because the old struct is only going to be used during the database migration.

### CODE REVIEW

File: beacon_node/beacon_chain/src/schema_change.rs [hunk 1]
```diff
@@ -73,6 +75,28 @@ pub fn migrate_schema<T: BeaconChainTypes>(
 
             Ok(())
         }
+        // Migration for adding `is_merge_complete` field to the fork choice store.
+        (SchemaVersion(4), SchemaVersion(5)) => {
```
Comment (michaelsproul) [inline beacon_node/beacon_chain/src/schema_change.rs]: There's a schema 5 in `unstable` already for checkpoint sync, so this should be 5->6 once `merge-f2f` gets updated for the latest `unstable`

**Comment by paulhauner** [comment]
Rebasing onto `merge-f2f` should (hopefully) solve these merge conflicts pretty cleanly.

I suggest doing a `rebase -i merge-f2f` and then just deleting all the commits that aren't your own (i.e., anything before cbedda1)

**Comment by paulhauner** [comment]
Whilst working on #2635, I noticed that we need access to the execution block hash for finalized block when we're [setting the head](https://github.com/ethereum/consensus-specs/blob/dev/specs/merge/fork-choice.md#notify_forkchoice_updated). We don't have this on hand, so there's two options:

1. Load the finalized block from the DB, then read the execution payload block hash from that (I'm doing this for the time being).
1. Store the payload block hash on each block in the fork choice.

Going with (2) seems appealing. Whilst (1) only involves block read (not a state read) from the database, it's still IO that I think would be nice to avoid.

Instead of storing the `is_merge_complete` bool, we could store the `execution_payload.block_hash` and then say that `is_merge_complete = execution_payload.block_hash != Hash256::zero()`.

What are your thoughts @realbigsean? Sorry for the last minute info.

**Comment by realbigsean** [comment]
@paulhauner I rebased as you suggested, but I'm a little confused because I'm seeing https://github.com/sigp/lighthouse/pull/2627/commits/a7e866fe4907fbd23aec6e77b9738411e99cb0b9 show up in this PR's diff but it already should be in the `merge-f2f` branch?

I implemented (2) here, it wasn't much to change on my end and it seems reasonable to me :+1:

**Comment by realbigsean** [comment]
Moving to #2643

**Pull request closed without merging**